accepted/0000-hub_retail.md

```diff
@@ -0,0 +1,213 @@
+Central creation of groups is covered in [Hub Users RFC](00061-hub-users.md). The salt states
+can be extended to handle also formula data.
+This approach is probably more flexible.
+
+Both alternatives will have to cope with authentization and Hub vs. Peripheral server user credentials.
+
+YAML configuration
+------------------
+
+FIXME: optional, related to Alternative 2 in previous section
+
+We have retail_yaml tool for initial configuration of Retail environment from YAML file.
+[Hub Users RFC](00061-hub-users.md) introduced similar concept, based on Salt.
+Maybe these parts could be merged.
+
+
+Combined SUMA and Branch server
+-------------------------------
+
+Branch services are configured by Salt formulas. Also PXE entries for each terminal are generated
+by Salt (currently via reactor).
+With combined SUMA and Branch server, these formulas should be applied on SUMA server itself.
+
+Terminals connect to SUMA directly, SUMA Proxy is not needed.
+
+
+FIXME: pick one alternative
+
+Alternative 1: Hub as a Salt Master for Combined SUMA and Branch server
+
+- this is already used for [Hub Users RFC](00061-hub-users.md)
+
+problems:
+- does not work for PXE entries and disconnected operation
+- mixed environment with Hub, Peripheral Servers with standalone Branch Serveres and 
+  Combined Peripheral Servers will have non-optimal configuration
+- does not cover the "Single server" setup
+
+
+Alternative 2: Multi-master setup of Combined SUMA and Branch server
+
+- Salt minion running on Combined SUMA and Branch server has 2 masters - Hub and itself
+
+problems:
+- SUMA registered to itself cause many problems
+  - we already have clients that do not allow certain actions (containers) so this should be fixable
+- Salt Multi-master might not be reliable
+
+
+Alternative 3: Use salt-call --local for Branch server part of combined server
+
+- how to represent such setup in GUI and API ?
+- would need significant changes in SUMA code
+
+
+Alternative 4: Virtualize Branch server part of combined server
+
+- use separate VM / container for the Branch server
+- the minion connects to SUMA on the same physical host
+
+- solves all the problems of other alternatives
+- Branch server as VM could be solved by documentation only - the user might be
+aready using some virtualization setup and we don't want to break it.
+
+
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Why should we **not** do this?
+
+  * obscure corner cases
+  * will it impact performance?
+  * what other parts of the product will be affected?
+  * will the solution be hard to maintain in the future?
+
+# Alternatives
+[alternatives]: #alternatives
+
+
+Images + Image metadata:
+
+- Implement Kiwi Image Store - HTTP server + API to handle metadata (similar functionality as Docker store)
+
+- Work with Kiwi upstream to include all the data to the image tarball, so the import functionality can be simplified
+
+- Use of Bittorrent protocol to distribute images from Hub to branches
```

Comment:
Should branch servers be registered directly to a hub server? Or we are talking about peripheral servers as branch server?

Reply:
I see this as that Hub where images are stored will run bittorrent tracker. Image pillar then can just include magnet link and branches/peripheral servers will download it. As long as other branches/peripheral servers are also seeding the images this can help with lowering the strain on hub to distribute the images. Alternative to http download used by image-sync state.

Reply:
I'd leave the transport choice up to the ISS implementation and yes, I also see Bittorrent as a possibility.